comit/src/network/swap_digest.rs

```diff
@@ -1,19 +1,42 @@
 use crate::{asset, identity, network::protocols::announce::SwapDigest, RelativeTime, Timestamp};
-use digest::Digest;
+use digest::{Digest, ToDigestInput};
+
+// TODO should we use swap_protocols::SwapProtocol instead?
+#[derive(Clone, Digest, Debug, PartialEq, Copy)]
+#[digest(hash = "SwapDigest")]
+pub enum SwapProtocol {
+    #[digest(prefix = "4b17")]
+    Hbit,
+    #[digest(prefix = "e5ec")]
+    Herc20,
+    #[digest(prefix = "c3d3")]
+    Halight,
+}
+
```

Comment:
We do not need this new type, we have a `Protocol` enum in `comit/src/lib.rs`.  You just need to add to that file
```
impl ToDigestInput for Protocol {
    fn to_digest_input(&self) -> Vec<u8> {
        let s = format!("{}", self);
        s.into_bytes()
    }
}
```
or some cleaner version of that.

Reply:
And then you can remove the TODO line 16 https://github.com/comit-network/comit-rs/pull/2723/files#r430780519